libvast/include/vast/type.hpp

```diff
@@ -1214,6 +1214,24 @@ class record_type final : public stateful_type_base {
       // nop
     }
 
+    template <type_or_concrete_type Type>
+    basic_field(String n, const Type& t,
+                const std::vector<type::attribute_view>& attrs) noexcept
+      : name{std::move(n)}, type(t, attrs) {
+      // nop
+    }
+
+    template <type_or_concrete_type Type>
+    basic_field(String n, const Type& t,
+                const std::vector<std::string>& attrs) noexcept
+      : name{std::move(n)} {
+      std::vector<type::attribute_view> transformed;
+      transformed.reserve(attrs.size());
+      for (auto const& attr : attrs)
+        transformed.push_back(type::attribute_view{attr});
+      type = vast::type(t, std::move(transformed));
+    }
```

Comment:
This breaks the mental model: The attribute belongs to the type, not to the field.

Reply:
Ok, removed it again